core/templates/dev/head/pages/exploration_player/AudioTranslationManagerService.js

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to manage the current language being
+ * used for audio translations as well as related modals.
+ */
+
+oppia.factory('AudioTranslationManagerService', [
+  '$modal', 'ExplorationPlayerStateService', 'AudioPlayerService',
+  function(
+      $modal, ExplorationPlayerStateService, AudioPlayerService) {
+    var _currentAudioLanguageCode = null;
+    var _allLanguageCodesInExploration = null;
+
+    var _init = function() {
+      _allLanguageCodesInExploration =
+        ExplorationPlayerStateService.getAllAudioLanguageCodes();
+
+      // TODO(tjiang11): Define an order. Preferably, we'd want
```

Comment:
We should keep track of these todos somewhere outside the code -- either in the milestones doc or in the github tracker. TODOs left in code get stale.